### ISSUE TITLE
Character Format

### ISSUE THREAD

**Issue by Sazu-bit** [LostRuins/koboldcpp#689]
I've been using some guidance regarding character formatting from SillyTavern, but I still seem to be having problems with keeping the AI from going cuckoo. I tried recently Airoboros 13B thinking that it might help, but it's worse than MythoMax 13B so I've returned to that model. 

When I say that the AI is going cuckoo, I mean that it keeps losing words. For example it will start skipping pronouns, including I, We, She, Him, They... sometimes it skips prepositions and I suspect that these are down to the PList format. 

Currently I use something like 

    [Samantha's persona: charismatic, funny, leader, feathered cap, believes in self-improvement/independence, values friendship/courage, friends(Toby, Chrissa, Raphael), mother is Helen, father is Frederick, no romantic relationship; Samantha's personality: logical, compassionate, patient; Samantha's appearance: age(35), gender(female), height(178cm, tall), hair(long blonde), skin(dark), body-shape(muscular), often wears cargo-pants/webbing; Samantha's background: raised by her parents in a wealthy household, went on to attend a boarding school before enlisting with the military. Samantha was commissioned as an officer and commanded her own platoon known as the Fire Devils; Scenario: Samantha meets {{user}} shortly after her platoon is decimated after information is leaked by a traitor to the enemy]

Despite ticking "always present" the AI will seemingly forget the gender, skip the pronouns entirely, confuse user as the Father etcetera, despite much correcting. I typically put this information in the World Info as I use Memory for a more wider view, while using the Author's note as a current events tracker. 

I've tried to use example sentences in the World Info in the same section as each character using 

    <START>
    {{user}}: Some speech *Actions in the first person*
    Samantha: Some speech *Actions in the third person*
    
But this again seems to confuse the AI, not that it won't follow the guidance per se but mixing the formats confuses the hell out of it and I'm not sure how to apply the PLIST format to the example sentences (I'm not even sure they're in the right place).

**TL;DR**

I'm looking for the compatibility documents for KoboldCPP into which format that it prefers. The guide that I have followed generally is: https://rentry.co/kingbri-chara-guide, but I've also looked at:

    Trappu's PLists + Ali:Chat guide: https://wikia.schneedc.com/bot-creation/trappu/creation
    AliCat's Ali:Chat guide: https://rentry.co/alichat
    kingbri's minimalistic guide: https://rentry.co/kingbri-chara-guide
    Kuma's W++ guide: https://rentry.co/WPP_For_Dummies"
    
I don't use Ali:Chat at present. Notably I'm not using SillyTavern for my character at the moment, I'm using KoboldLite as packaged with KoboldCpp. 

Version is KoboldCpp 1.53, I can't use the later versions due to https://github.com/LostRuins/koboldcpp/issues/672, otherwise I would happily upgrade.

**Comment by Sazu-bit** [comment]
Attempting to use a work around (https://libreddit.strongthany.cc/r/SillyTavernAI/comments/15t2bkf/mythomax_13b/) that suggests using Mirostat (mode 2, 5, 0.1) hopefully that will reduce that problem. I'll also be reducing my cores from 7 (8 available) to 5 as this might have an impact as well. Keeping here as I'm not actually using Instruct at the moment (mostly because it keeps outputting ### Instruct, ### Output etc, which is not immersive for me).

**Comment by Woisek** [comment]
I'm not sure on how exactly I can help you, but all I can say is: You really, really should look into and learn how to use AliChat:PList format for creating your bots and learn all of Trappu's resources/tutorials. In my opinion, forget other writing styles, because it seems, that AliChat:PList is the most efficient way to create a bot for local models.
What you use (the example you gave) is almost PList already, so it would only be logical to do the final step for it. Also, I would highly encourage to use SillyTavern as frontend and kobaldccp as the model loader (ooba is nice, but a bit too bloated, I switched long ago).

Your problem with the "missing words" has nothing to do with PList or any writing of the bot, but with the repetition penalty. If it's too high, then common words will be dropped, IIRC.
Using the Mirostat settings is also a good idea.

There are sure more optimizing settings, but that's what I can say right now to this, I hope it helps a bit.

**Comment by LostRuins** [comment]
I would suggest looking at some example cards or characters and working from there. The built-in scenarios in Lite are quite good.

**Comment by Sazu-bit** [comment]
I've looked at various cards, but I often find that they are... very poor with very little standardization. I prefer making my own as I have more control over them and I can fully use the SillyTavern functionality that way. 

For now, I've activated Mirostat (Version 2) and I've lowered the repetition penalty to 1.18, occasionally I have problems with the bot repeating certain phrases and the various tips suggested using a higher repetition penalty to combat such things. Thanks to your advice, I don't seem to be having the same problem. 

Am I to understand that KoboldCpp itself doesn't have a preference and that this might be down to the model?

Is there a guide on using the Instruct Mode for role-playing? I am under the impression that this might yield even better results. MythoMax for example may prefer the Instruct method but I've yet to come across suitable information.

**Comment by Woisek** [comment]
The koboldccp.exe is a small standalone program that loads a model. The only settings it has, is for loading layers, setting contextsize and such things. All generation settings are done in the used frontend like SillyTavern.
So, if you used SillyTavern, you will find tons of help for settings on their Discord (https://discord.gg/sillytavern).
I would suggest giving it a try.

**Comment by LostRuins** [comment]
Yes there are many bad cards out there, but there are some good ones too. Koboldcpp's preference is the defaults for Lite, which is using top_p=0.9, temp=0.7 and rep_pen=1.1, everything else is disabled.
